chain/chain/src/store.rs

```diff
@@ -1574,6 +1574,8 @@ impl<'a> ChainStoreUpdate<'a> {
         Ok(())
     }
 
+    /// This function checks that the block is not on a chain with challenged blocks and updates
```

Comment:
wait -- but you just removed the check for challenged blocks from here, right ?

Reply:
Oh no sorry I didn't mean to remove the checks here because they are still needed in header sync when processing headers. (I could call the other function again together with this function, but I'm afraid that that's going to make header sync take even longer and I don't have time to debug why header sync takes so long right now, so I decided to simply keep the code the same here).

I'll revert the change here.